### PR TITLE
Convert Eclipse from Docbook to CommonMark

### PR DESCRIPTION
**Pull request by ngiger**
###### Motivation for this change

Converted xml docs to md docs as per issue #105251

###### Things done

Converted xml to md and updated index.xml to reference it.

### CODE REVIEW

File: doc/builders/packages/eclipse.section.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+function `eclipseWithPlugins` found inside the `nixpkgs.eclipses`
+attribute set. This function takes as argument
+`{ eclipse, plugins ? [], jvmArgs ? [] }` where `eclipse` is a one of
+the Eclipse packages described above, `plugins` is a list of plugin
+derivations, and `jvmArgs` is a list of arguments given to the JVM
+running the Eclipse. For example, say you wish to install the latest
+Eclipse Platform with the popular Eclipse Color Theme plugin and also
+allow Eclipse to use more RAM. You could then add
+
+```ShellSession
```
Comment (jtojnar) [inline doc/builders/packages/eclipse.section.md:36]: This is nix code.

File: doc/builders/packages/eclipse.section.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  }
+```
+
+to your Nixpkgs configuration (`~/.config/nixpkgs/config.nix`) and
+install it by running `nix-env -f '<nixpkgs>' -iA myEclipse` and
+afterward run Eclipse as usual. It is possible to find out which plugins
+are available for installation using `eclipseWithPlugins` by running
+
+```ShellSession
+    $ nix-env -f '<nixpkgs>' -qaP -A eclipses.plugins --description
```
Comment (jtojnar) [inline doc/builders/packages/eclipse.section.md:52]: This should not be indented.

**Review by jtojnar — COMMENTED** [review]
Could you also format the file using prettier as described in https://github.com/NixOS/nixpkgs/issues/105243?

File: doc/builders/packages/eclipse.section.md (hunk 1 above)
Comment (ngiger) [reply doc/builders/packages/eclipse.section.md:36]: Sorry. I am no markdown specialist.

Should I replace ShellSession by nix?

Comment (jtojnar) [reply doc/builders/packages/eclipse.section.md:36]: Yes. And same for the other Nix code.

**Comment by ngiger** [comment]
What do I have to do to get this pull request merged?

**Comment by jtojnar** [comment]
The review points still have not been resolved.

**Comment by taku0** [comment]
Please run the following command to run `prettier`:

```
nix-shell -I nixpkgs=channel:nixos-unstable -p nodePackages.prettier --run 'prettier --prose-wrap never --write doc/builders/packages/eclipse.section.md'
```

**Pull request merged**